compiler/src/iree/compiler/Codegen/Interfaces/PartitionableLoopsInterface.cpp

```diff
@@ -160,11 +165,12 @@ struct AllParallelAsPartitionableLoops
       }
       partitionableLoops.push_back(iteratorType.index());
     }
-    if (partitionableLoops.size() > maxNumPartitionedLoops) {
-      partitionableLoops.erase(
-          partitionableLoops.begin(),
-          std::next(partitionableLoops.begin(),
-                    partitionableLoops.size() - maxNumPartitionedLoops));
+    if (maxNumPartitionedLoops.has_value() &&
```

Comment:
(and here as well)
Style nit. Just the indentations gotten a bit unreadable. Maybe flip the condition, so
```
if (!maxNumPartitionableLoops.has_value() || partitionableLoops.size() <= maxNumPartitionedLoops.value()) {
  return partitionableLoops;
}
partitionableLoops.erase(...)

```